plutus-pab/src/Plutus/PAB/Core/ContractInstance/STM.hs

```diff
@@ -153,19 +157,23 @@ data BlockchainEnv =
         , beCurrentSlot         :: TVar Slot -- ^ Actual current slot
         , beLastSyncedBlockSlot :: TVar Slot -- ^ Slot of the last synced block from 'startNodeClient'
         , beLastSyncedBlockNo   :: TVar BlockNumber -- ^ Last synced block number from 'startNodeClient'.
-        , beTxChanges           :: TVar (UtxoIndex TxIdState) -- ^ Map holding metadata which determines the status of transactions.
+        , beTxChanges           :: Either (TVar (UtxoIndex TxIdState)) (IORef TCSIndex)-- ^ Map holding metadata which determines the status of transactions.
         , beTxOutChanges        :: TVar (UtxoIndex TxOutBalance) -- ^ Map holding metadata which determines the status of transaction outputs.
```

Comment:
Shouldn't the same be done with `UtxoIndex TxOutBalance`? This also uses a lot of memory

Reply:
Actually it does not use that much memory. I would say, let's leave it like this for now as this one is a fairly large commit already and add another issue if you feel we should do that.

Reply:
Aren't the changes to `beTxOutChanges` pretty much the exact same as `beTxChanges`?

Reply:
They will need a new indexer. I don't think it's a bad idea, but maybe we should have a separate issue for it.

Reply:
Will add a ticket